File: bibfor/supervis/debut.f
      SUBROUTINE DEBUT  ( IPASS,IER )
      IMPLICIT NONE
      INTEGER                 IER,IPASS
C     ------------------------------------------------------------------
C            CONFIGURATION MANAGEMENT OF EDF VERSION
C MODIF SUPERVIS  DATE 21/09/2011   AUTEUR COURTOIS M.COURTOIS 
C ======================================================================
C COPYRIGHT (C) 1991 - 2011  EDF R&D                  WWW.CODE-ASTER.ORG
C THIS PROGRAM IS FREE SOFTWARE; YOU CAN REDISTRIBUTE IT AND/OR MODIFY
C IT UNDER THE TERMS OF THE GNU GENERAL PUBLIC LICENSE AS PUBLISHED BY
C THE FREE SOFTWARE FOUNDATION; EITHER VERSION 2 OF THE LICENSE, OR
C (AT YOUR OPTION) ANY LATER VERSION.
C
C THIS PROGRAM IS DISTRIBUTED IN THE HOPE THAT IT WILL BE USEFUL, BUT
C WITHOUT ANY WARRANTY; WITHOUT EVEN THE IMPLIED WARRANTY OF
C MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. SEE THE GNU
C GENERAL PUBLIC LICENSE FOR MORE DETAILS.
C
C YOU SHOULD HAVE RECEIVED A COPY OF THE GNU GENERAL PUBLIC LICENSE
C ALONG WITH THIS PROGRAM; IF NOT, WRITE TO EDF R&D CODE_ASTER,
C    1 AVENUE DU GENERAL DE GAULLE, 92141 CLAMART CEDEX, FRANCE.
C ======================================================================
C TOLE CRP_7
C    DECODAGE DE LA COMMANDE DEBUT OU POURSUITE
C     ------------------------------------------------------------------
C     ROUTINE(S) UTILISEE(S) :
C        IBBASE  IBCATA
C     ------------------------------------------------------------------
C
      INTEGER    MXCMD
      PARAMETER (MXCMD = 500 )
      CHARACTER*8 K8B, REPONS
      CHARACTER*16 NOMCMD, K16B, CMPDEF, CMPUT, CMPOUT
      CHARACTER*80 FICHDF
      INTEGER IERIMP, ICMD, LOUT, N, NCODE
      INTEGER      IARG
C
      FICHDF=' '
C
C --- ERREUR / ERREUR_F :
C     MOT-CLE CODE PRESENT ?
      CALL GETFAC('CODE', NCODE)
      IF ( NCODE .NE. 0 ) THEN
         CMPDEF = 'ABORT'
      ELSE
         CMPDEF = 'EXCEPTION'
C        FERMETURE DU .CODE (OUVERT PAR IB0MAI)
         CALL ULOPEN(-15, ' ', ' ', ' ', ' ')
      ENDIF
      CALL GETVTX('ERREUR', 'ERREUR_F', 1,IARG, 1, CMPUT, N)
      IF ( N .EQ. 1 ) THEN
         CMPDEF = CMPUT
      ENDIF
      CALL ONERRF(CMPDEF, CMPOUT, LOUT)
C
      IER    = 0
      IF(IPASS .NE. 1) GOTO 9999
         CALL IBIMPR( IERIMP )
         CALL PRINIT()
         CALL PRENTE()
C --- LECTURE DU MOT CLE FACTEUR DEBUG OU DE GESTION MEMOIRE DEMANDE
      CALL IBDBGS()
C
C --- ALARME GENERIQUE
      CALL GETVTX(' ','PAR_LOT',1,IARG,1,REPONS,N)
      IF ( REPONS .EQ. 'NON' ) THEN
         CALL U2MESS('A', 'SUPERVIS_1')
      ENDIF
C
C --- LECTURE DU MOT CLEF TEMPS_CPU
C
      IF ( IER .EQ. 0 ) CALL IBTCPU (IER)
C
C --- LECTURE DU MOT CLE HDF ---
C
      IF ( IER .EQ. 0 ) CALL IBFHDF( IER , FICHDF )
C
C --- LECTURE DU MOT CLE FACTEUR BASE ET ---
C --- ALLOCATION DES BASES DE DONNEES ---
      IF ( IER .EQ. 0 ) CALL IBBASE( IER , FICHDF )
      IF ( IER. EQ. 0 ) THEN
         CALL GETRES(K8B,K16B,NOMCMD)
C        -- INITIALISATION DE LA FONCTION NULLE : '&FOZERO'
C           ET DU COMMON FOSAV
         CALL FOZERO('&FOZERO')
         CALL FOINT0
      ENDIF
C
C --- POUR EVITER QUE LA CREATION DE '&&_NUM_CONCEPT_UNIQUE'
C        NE SOIT REPROCHE A UNE COMMANDE CREANT UNE SD
C        (DEBUT/DEBUG/SDVERI='OUI')
      CALL GCNCON('.',K8B)
C
C --- INITIALISATION DE LA TABLE DES CONCEPTS ---
      CALL GCUINI( MXCMD , 'G' , IER )
C
C --- LECTURE DU MOT CLE FACTEUR  CATALOGUE ---
      IF ( IER .EQ. 0 .AND. FICHDF .EQ. '  ') CALL IBCATA( IER )
C
C --- STATS SUR LA COMMANDE DE DEMARRAGE  ---
C
      IF ( IER .EQ. 0 ) CALL GCUOPR( 1, ICMD )
 9999 CONTINUE
      END

File: code_aster/Cata/Commands/debut.py
# coding=utf-8
# ======================================================================
# COPYRIGHT (C) 1991 - 2017  EDF R&D                  WWW.CODE-ASTER.ORG
# THIS PROGRAM IS FREE SOFTWARE; YOU CAN REDISTRIBUTE IT AND/OR MODIFY
# IT UNDER THE TERMS OF THE GNU GENERAL PUBLIC LICENSE AS PUBLISHED BY
# THE FREE SOFTWARE FOUNDATION; EITHER VERSION 2 OF THE LICENSE, OR
# (AT YOUR OPTION) ANY LATER VERSION.
#
# THIS PROGRAM IS DISTRIBUTED IN THE HOPE THAT IT WILL BE USEFUL, BUT
# WITHOUT ANY WARRANTY; WITHOUT EVEN THE IMPLIED WARRANTY OF
# MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. SEE THE GNU
# GENERAL PUBLIC LICENSE FOR MORE DETAILS.
#
# YOU SHOULD HAVE RECEIVED A COPY OF THE GNU GENERAL PUBLIC LICENSE
# ALONG WITH THIS PROGRAM; IF NOT, WRITE TO EDF R&D CODE_ASTER,
#    1 AVENUE DU GENERAL DE GAULLE, 92141 CLAMART CEDEX, FRANCE.
# ======================================================================
# person_in_charge: j-pierre.lefebvre at edf.fr
from code_aster.Cata.Syntax import *
from code_aster.Cata.DataStructure import *
from code_aster.Cata.Commons import *


DEBUT=MACRO(nom="DEBUT",
            op=OPS("code_aster.Cata.ops.build_debut"),
            repetable='n',
            fr=tr("Ouverture d'une étude. Allocation des ressources mémoire et disque et fichiers"),
            sd_prod=ops.DEBUT,

         PAR_LOT         =SIMP(fr=tr("mode de traitement des commandes"),statut='f',typ='TXM',
                           into=("OUI","NON"),defaut="OUI"),
         IMPR_MACRO      =SIMP(fr=tr("affichage des sous-commandes produites par les macros dans le fichier mess"),
                           statut='f',typ='TXM',into=("OUI","NON"),defaut="NON"),
#         FORMAT_HDF      =SIMP(fr=tr("sauvegarde de la base GLOBALE au format HDF"),statut='f',
#                               typ='TXM',defaut="NON",into=("OUI","NON",) ),
         BASE            =FACT(fr=tr("définition des paramètres associés aux bases JEVEUX"),
                               statut='f',min=1,max=2,
           FICHIER         =SIMP(fr=tr("nom de la base"),statut='o',typ='TXM',
                                 into=('GLOBALE','VOLATILE'),),
           TITRE           =SIMP(statut='f',typ='TXM'),
           CAS             =SIMP(statut='f',typ='TXM'),
           NMAX_ENRE       =SIMP(fr=tr("nombre maximum d enregistrements"),statut='f',typ='I'),
           LONG_ENRE       =SIMP(fr=tr("longueur des enregistrements"),statut='f',typ='I'),
           LONG_REPE       =SIMP(fr=tr("longueur du répertoire"),statut='f',typ='I'),
           TAILLE          =SIMP(statut='c', typ='I', fr=tr("ne pas utiliser")),
         ),

         CATALOGUE       =FACT(statut='f',min=1,max=10,
           FICHIER         =SIMP(statut='o',typ='TXM'),
           UNITE           =SIMP(statut='f',typ=UnitType(),inout='in'),
         ),

         CODE            =FACT(fr=tr("paramètres réservés aux cas-tests"),
                               statut='f',min=1,max=1,
           NIV_PUB_WEB     =SIMP(statut='o',typ='TXM',into=('INTERNET','INTRANET')),
           VISU_EFICAS     =SIMP(statut='f',typ='TXM',into=('OUI','NON'),defaut='OUI'),
         ),

         ERREUR          =FACT(fr=tr("comportement en cas d'erreur"),statut='f',min=1,max=1,
           ERREUR_F        =SIMP(statut='f',typ='TXM',into=('ABORT','EXCEPTION'),),
         ),

         DEBUG           =FACT(fr=tr("option de déboggage reservée aux développeurs"),
                               statut='d',min=1,max=1,
           JXVERI          =SIMP(fr=tr("vérifie l intégrité de la segmentation mémoire"),
                                 statut='f',typ='TXM',into=('OUI','NON'),defaut='NON'),
           SDVERI          =SIMP(fr=tr("vérifie la conformité des SD produites par les commandes"),
                                 statut='f',typ='TXM',into=('OUI','NON'), defaut='NON'),
           JEVEUX          =SIMP(fr=tr("force les déchargement sur disque"),
                                 statut='f',typ='TXM',into=('OUI','NON'),defaut='NON'),
           ENVIMA          =SIMP(fr=tr("imprime les valeurs définies dans ENVIMA"),
                                 statut='f',typ='TXM',into=('TEST',)),
           HIST_ETAPE = SIMP(fr=tr("permet de conserver toutes les étapes du jeu de commandes"),
                             statut='f', typ='TXM', into=('OUI', 'NON'), defaut='NON'),
         ),

         MESURE_TEMPS     =FACT(fr=tr("Pour afficher le temps des principales étapes de calcul"),
                               statut='d',min=1,max=1,
           NIVE_DETAIL      =SIMP(fr=tr("niveau de détail des impressions"),
                                 statut='f',typ='I',into=(0,1,2,3),defaut=1),
                                 # 0 : rien
                                 # 1 : impression en fin de commande des mesures principales
                                 # 2 : impression en fin de commande des mesures principales et secondaires
                                 # 3 : impression des mesures principales et secondaires pour chaque pas de temps
           MOYENNE     =SIMP(fr=tr("affichage des moyennes et écart-types en parallèle"),
                                  statut='f',typ='TXM',into=('OUI','NON',),defaut='NON'),
         ),

         MEMOIRE         =FACT(fr=tr("mode de gestion mémoire utilisé"),statut='d',min=1,max=1,
           TAILLE_BLOC       =SIMP(statut='f',typ='R',defaut=800.),
           TAILLE_GROUP_ELEM =SIMP(statut='f',typ='I',defaut=1000),
         ),

         RESERVE_CPU     =FACT(fr=tr("reserve de temps pour terminer une execution"),statut='d',max=1,
           regles=(EXCLUS('VALE','POURCENTAGE'),),
#          par défaut VALE fixée à 10. dans le FORTRAN si CODE présent
           VALE            =SIMP(statut='f',typ='I',val_min=0,),
#          par défaut 10% dans le FORTRAN
           POURCENTAGE     =SIMP(statut='f',typ='R',val_min=0.,val_max=1.0),
#          valeur en secondes de la réserve maximum bornée à 900 secondes
           BORNE           =SIMP(statut='f',typ='I',val_min=0,defaut=900),),

         IGNORE_ALARM = SIMP(statut='f', typ='TXM', max='**', fr=tr("Alarmes que l'utilisateur souhaite délibérément ignorer")),

         LANG = SIMP(statut='f', typ='TXM',
                     fr=tr("Permet de choisir la langue utilisée pour les messages (si disponible)"),
                     ),

         INFO     = SIMP(statut='f', typ='I', defaut=1, into=(1,2),),
);
